google/cloud/security/scanner/audit/instance_network_interface_rules_engine.py

```diff
@@ -0,0 +1,251 @@
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Rules engine for NetworkInterface."""
+from collections import namedtuple
+import itertools
+import re
+
+# pylint: enable=line-too-long
+from google.cloud.security.common.util import log_util
+from google.cloud.security.scanner.audit import base_rules_engine as bre
+from google.cloud.security.scanner.audit import errors as audit_errors
+
+LOGGER = log_util.get_logger(__name__)
+
+
+def escape_and_globify(pattern_string):
+    """Given a pattern string with a glob, create actual regex pattern.
+
+    To require > 0 length glob, change the "*" to ".+". This is to handle
+    strings like "*@company.com". (THe actual regex would probably be
+    ".*@company.com", except that we don't want to match zero-length
+    usernames before the "@".)
+
+    Args:
+        pattern_string: The pattern string of which to make a regex.
+
+    Returns:
+    The pattern string, escaped except for the "*", which is
+    transformed into ".+" (match on one or more characters).
+    """
+    return '^{}$'.format(re.escape(pattern_string).replace('\\*', '.+'))
+
+
+class InstanceNetworkInterfaceRulesEngine(bre.BaseRulesEngine):
+    """Rules engine for InstanceNetworkInterfaceRules."""
+
+    def __init__(self, rules_file_path, snapshot_timestamp=None):
+        """Initialize.
+
+        Args:
+            rules_file_path: file location of rules
+        """
+        super(InstanceNetworkInterfaceRulesEngine,
+              self).__init__(rules_file_path=rules_file_path)
+        self.rule_book = None
+
+    def build_rule_book(self, global_configs=None):
+        """Build InstaceNetworkInterfaceRuleBook from rules definition file."""
+        self.rule_book = InstaceNetworkInterfaceRuleBook(
+            self._load_rule_definitions())
+
+    # pylint: disable=arguments-differ
+    def find_policy_violations(self, instance_network_interface,
+                               force_rebuild=False):
+        """Determine whether the networks violates rules."""
+        violations = itertools.chain()
+        if self.rule_book is None or force_rebuild:
+            self.build_rule_book()
+        resource_rules = self.rule_book.get_resource_rules()
+
+        for rule in resource_rules:
+            violations = itertools.chain(violations,
```

Comment:
nit: do you think it might work (line-length wise) to do the following, for readability?

```
    violations = itertools.chain(
        violations,
        rule.find_violations(instance_network_interface))
```